worker/control/handlers.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package control
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/environs/bootstrap"
+	"github.com/juju/juju/state"
+)
+
+const (
+	// jujuMetricsUserPrefix defines the "namespace" in which this worker is
+	// allowed to create/remove users.
+	jujuMetricsUserPrefix = "juju-metrics-"
+
+	// userCreator is the listed "creator" of metrics users in state.
+	userCreator = "admin"
+)
+
+func (w *Worker) registerHandlers(handle func(pattern string, handler http.Handler)) {
+	// TODO: does this API structure make sense?
+	handle("/metrics-users/add", http.HandlerFunc(w.handleAddMetricsUser))
+	handle("/metrics-users/remove", http.HandlerFunc(w.handleRemoveMetricsUser))
```

Comment:
It's probably more idiomatic to have

`POST /metrics-users` for creation and `DELETE /metrics-users/{user}` for deletion :)

Reply:
you're right, the only thing I dislike about `DELETE /metrics-users/{user}` is it implies there is a `metrics-users` namespace separate to the `users` namespace, when in fact we're just allowing deletion on a subset of `users`